nixos/modules/services/web-apps/mealie.nix

```
@@ -0,0 +1,89 @@
+{ config, lib, pkgs, ...}:
+let
+  cfg = config.services.mealie;
+  pkg = cfg.package;
+
+  # TODO Fix ingredient quantities NLP recognition
+in
+{
+  options.services.mealie = {
+    enable = lib.mkEnableOption "Mealie, a recipe manager and meal planner";
+
+    package = lib.mkPackageOption pkgs "mealie" { };
+
+    listenAddress = lib.mkOption {
+      type = lib.types.str;
+      default = "0.0.0.0";
+      description = "Address on which the service should listen.";
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 9000;
+      description = "Port on which to serve the Mealie service.";
+    };
+
+    settings = lib.mkOption {
+      type = lib.types.submodule {
+        freeformType = with lib.types; attrsOf str;
+      };
+      default = {};
+      description = lib.mdDoc ''
+        Configuration of the Mealie service.
+
+        See [the mealie documentation](https://nightly.mealie.io/documentation/getting-started/installation/backend-config/) for available options and default values.
+
+        In addition to the official documentation, you can set {env}`MEALIE_LOG_FILE`.
+      '';
+      example = {
+        ALLOW_SIGNUP = "false";
+      };
+    };
+
+    credentialsFile = lib.mkOption {
+      type = with lib.types; nullOr path;
+      default = null;
+      example = "/run/secrets/mealie-credentials.env";
+      description = ''
+        File containing credentials used in mealie such as {env}`POSTGRES_PASSWORD`
+        or sensitive LDAP options.
+
+        Expects the format of an `EnvironmentFile=`, as described by {manpage}`systemd.exec(5)`.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.mealie = {
+      description = "Mealie, a self hosted recipe manager and meal planner";
+      after = [
+        "network.target"
+        "network-online.target"
+      ];
+      wants = [
+        "network-online.target"
+      ];
+      wantedBy = [
+        "multi-user.target"
+      ];
+
+      environment = {
+        PYTHONPATH = pkg.python.pkgs.makePythonPath ([ pkg ] ++ pkg.propagatedBuildInputs);
+        PRODUCTION = "true";
+        MEALIE_LOG_FILE = "/var/log/mealie/mealie.log";
+        ALEMBIC_CONFIG_FPATH="${pkg}/config/alembic.ini";
+        API_PORT = toString cfg.port;
+      } // (builtins.mapAttrs (_: val: toString val) cfg.settings);
+
+      serviceConfig = {
+        DynamicUser = true;
+        User = "mealie";
+        ExecStartPre = "${pkg.python.interpreter} ${pkg}/bin/init_db.py";
+        ExecStart = "${pkg}/bin/start-mealie -b ${cfg.listenAddress}:${builtins.toString cfg.port}";
+        EnvironmentFile = lib.optional (cfg.credentialsFile != null) cfg.credentialsFile;
```

It's probably better to not set the option at all than to an empty list.

```suggestion
        EnvironmentFile = lib.mkIf (cfg.credentialsFile != null) cfg.credentialsFile;
```